settings/Controller/CorsController.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * @author Noveen Sachdeva "noveen.sachdeva@research.iiit.ac.in"
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+namespace OC\Settings\Controller;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\RedirectResponse;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\ILogger;
+use OCP\IRequest;
+use OCP\IURLGenerator;
+use OCP\IConfig;
+use OCP\IUserSession;
+
+/**
+ * This controller is responsible for managing white-listed domains for CORS
+ *
+ * @package OC\Settings\Controller
+ */
+class CorsController extends Controller {
+	/** @var ILogger */
+	private $logger;
+
+	/** @var IURLGenerator */
+	private $urlGenerator;
+
+	/** @var string */
+	private $userId;
+
+	/** @var IConfig */
+	private $config;
+
+	/**
+	 * CorsController constructor.
+	 *
+	 * @param string $AppName The app's name.
+	 * @param IRequest $request The request.
+	 * @param IUserSession $userSession Logged in user's session
+	 * @param ILogger $logger The logger.
+	 * @param IURLGenerator $urlGenerator Use for url generation
+	 * @param IConfig $config
+	 */
+	public function __construct($AppName, IRequest $request,
+								IUserSession $userSession,
+								ILogger $logger,
+								IURLGenerator $urlGenerator,
+								IConfig $config) {
+		parent::__construct($AppName, $request);
+
+		$this->AppName = $AppName;
+		$this->config = $config;
+		$this->userId = $userSession->getUser()->getUID();
+		$this->logger = $logger;
+		$this->urlGenerator = $urlGenerator;
+	}
```

Comment:
Can use properties promotion:
```suggestion
	public function __construct(
		$AppName, IRequest $request,
		IUserSession $userSession,
		private LoggerInterface $logger,
		private IURLGenerator $urlGenerator,
		private IConfig $config
	) {
		parent::__construct($AppName, $request);
		$this->userId = $userSession->getUser()->getUID();
	}
```
Also replace deprecated ILogger